### PR TITLE
Initial integration test implementation

### PR DESCRIPTION
**Pull request by chrisdunelm**
This compares the monolithic-generator and micro-generator outputs and only allows whitespace and trailing-comma changes.

Only a single generated file is currently compared; future PRs will add all other generated files, and other proto inputs.

### CODE REVIEW

**Comment by chrisdunelm** [comment]
Apologies that this is a moderately large PR.
However, all the changes to the test files are purely mechanical.

File: src/Main.php [hunk 1]
```diff
@@ -24,15 +24,33 @@
...
 // Generate PHP code.
 $year = (int)date('Y');
-$files = CodeGenerator::generateFromDescriptor($descBytes, $package, $year);
-foreach ($files as [$relativeFilename, $fileContent]) {
-    // TODO: Later this won't just print out the generated file content.
-    print("File: '{$relativeFilename}':\n");
-    print($fileContent . "\n");
+$files = CodeGenerator::generateFromDescriptor($descBytes, $package, $year, null);
+if (is_null($outputDir)) {
+    // TODO: Remove printout; only save files to the specified output path.
```
Comment (jskeet) [inline src/Main.php:45]: But isn't this comment for the situation where we don't *have* an output path specified? It seems like a reasonable feature to keep.

**Review by jskeet — APPROVED** [review]
As usual, I've only skimmed this, but it seems reasonable.

File: src/Main.php (hunk 1 above)
Comment (chrisdunelm) [reply src/Main.php:45]: Maybe... OK, I won't remove it yet :)

**Pull request merged**